src/widgets/modals/PatientEdit.js

```diff
@@ -0,0 +1,265 @@
+import React from 'react';
+import { ScrollView, View } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { ValidationTextInput } from '../ValidationTextInput';
+import { PageButton } from '../PageButton';
+
+import globalStyles, { WHITE, SUSSOL_ORANGE } from '../../globalStyles/index';
+
+import { PatientActions } from '../../actions/PatientActions';
+
+/**
+ * Component to be wrapped by `ModalContainer` acting as a form for adding or
+ * editing details of a patient.
+ *
+ */
+export const PatientEditComponent = ({
+  canSave,
+  setDateOfBirthValidity,
+  setAddressOneValidity,
+  setAddressTwoValidity,
+  setPatientCodeValidity,
+  setCountryValidity,
+  setFirstName,
+  setLastName,
+  setCode,
+  setDateOfBirth,
+  setEmail,
+  setPhone,
+  setAddressOne,
+  setAddressTwo,
+  setCountry,
+  firstName,
+  lastName,
+  code,
+  dateOfBirth,
+  email,
+  phone,
+  addressOne,
+  addressTwo,
+  country,
+  savePatient,
+}) => {
+  const { flexOne, flexTen, flexRow, buttonsRow, whiteBackground } = localStyles;
+  const { saveButton, saveButtonTextStyle, cancelButton, cancelButtonTextStyle } = localStyles;
+
+  const validateCode = React.useCallback(input => input.length < 20, []);
+  const validateCountry = React.useCallback(input => input.length < 50, []);
+  const validateAddress = React.useCallback(input => input.length < 50, []);
+  const validateDateOfBirth = React.useCallback(input => {
+    // Ensure the entered value is castable to a date and is less than
+    // the current date.
+    const dateTime = new Date(input).getTime();
+    const dateTimeNow = new Date().getTime();
+    if (input === '') return true;
+
+    const inputIsAValidDate = !Number.isNaN(dateTime);
+    const inputIsLessThanNow = dateTime < dateTimeNow;
+    const inputHasAForwardSlash = input.includes('/');
+
+    return inputIsAValidDate && inputIsLessThanNow && inputHasAForwardSlash;
+  });
+
+  return (
+    <View style={flexOne}>
+      <ScrollView style={whiteBackground}>
+        <View style={flexRow}>
+          <View style={flexOne} />
+          <View style={flexTen}>
+            <ValidationTextInput
+              value={firstName}
+              label="First Name"
+              isRequired
+              onSubmitEditing={setFirstName}
+            />
+            <ValidationTextInput
+              value={lastName}
+              isRequired
+              label="Last Name"
+              onSubmitEditing={setLastName}
+            />
+            <ValidationTextInput
+              value={code}
+              onValidate={validateCode}
+              invalidMessage="Must be less than 20 characters"
```

Comment:
not now: translation string for the validation messages

Reply:
Have made this issue to track all the localization changes to be done because I am lazy https://github.com/openmsupply/mobile/issues/1693